tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_bw_and_latency.cpp

```diff
@@ -0,0 +1,162 @@
+    uint32_t core_y = test_args::get_command_option_uint32(input_args, "-y", 0);
+    warmup_iterations_g = test_args::get_command_option_uint32(input_args, "-w", DEFAULT_WARMUP_ITERATIONS);
+    iterations_g = test_args::get_command_option_uint32(input_args, "-i", DEFAULT_ITERATIONS);
+    lazy_g = test_args::has_command_option(input_args, "-z");
+    time_just_finish_g = test_args::has_command_option(input_args, "-f");
+    source_mem_g = test_args::get_command_option_uint32(input_args, "-m", 0);
+    uint32_t size = test_args::get_command_option_uint32(input_args, "-s", DEFAULT_SIZE_K) * 1024;
+    latency_g = test_args::has_command_option(input_args, "-l");
+    page_size_g = test_args::get_command_option_uint32(input_args, "-p", DEFAULT_PAGE_SIZE);
+    page_count_g = size / page_size_g;
+
+    worker_g = {.start = {core_x, core_y}, .end = {core_x, core_y}};
+}
+
+int main(int argc, char **argv) {
+    init(argc, argv);
+
+    bool pass = true;
+    try {
+        int device_id = 0;
+        tt_metal::Device *device = tt_metal::CreateDevice(device_id);
+
+        CommandQueue& cq = tt::tt_metal::detail::GetCommandQueue(device);
+
+        tt_metal::Program program = tt_metal::CreateProgram();
+
+        string src_mem;
+        switch (source_mem_g) {
+        case 0:
+            src_mem = "FROM_PCIE";
+            break;
+        case 1:
+            src_mem = "FROM_DRAM";
+            break;
+        case 2:
+            src_mem = "FROM_L1";
+            break;
+        }
+
+        std::map<string, string> defines = {
+            {src_mem, "1"},
+            {"ITERATIONS", std::to_string(iterations_g)},
+            {"PAGE_SIZE", std::to_string(page_size_g)},
+            {"PAGE_COUNT", std::to_string(page_count_g)},
+            {"LATENCY", std::to_string(latency_g)}
+        };
+
+        tt_metal::CircularBufferConfig cb_config = tt_metal::CircularBufferConfig(page_size_g * page_count_g, {{0, tt::DataFormat::Float32}})
+            .set_page_size(0, page_size_g);
+        auto cb = tt_metal::CreateCircularBuffer(program, worker_g, cb_config);
+
+        auto dm0 = tt_metal::CreateKernel(
+                                          program,
+                                          "tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/kernels/bw_and_latency.cpp",
+                                          worker_g,
+                                          tt_metal::DataMovementConfig{.processor = tt_metal::DataMovementProcessor::RISCV_0, .noc = tt_metal::NOC::RISCV_0_default, .defines = defines});
+
+        // Cache stuff
+        for (int i = 0; i < warmup_iterations_g; i++) {
+            EnqueueProgram(cq, program, false);
+        }
+        Finish(cq);
+
+        if (lazy_g) {
+            tt_metal::detail::SetLazyCommandQueueMode(true);
+        }
+
+        auto start = std::chrono::system_clock::now();
+        EnqueueProgram(cq, program, false);
+        if (time_just_finish_g) {
```

Comment:
isn't this just dictated by whether or not we want to run in lazy mode? do we need a separate flag for this? I mean, I suppose we can measure the host writes in lazy mode as well, but don't see a use-case for doing that. Maybe you do?

Reply:
When I ran in lazy mode it ran slower, I wanted to tease out the up front cost from the execute cost.